backend/src/main/java/org/eclipse/tractusx/puris/backend/file/logic/service/ExcelService.java

```diff
@@ -785,11 +785,22 @@ private List<DataImportError> evaluateWorkbook(Workbook workbook) {
                             try {
                                 evaluator.evaluateFormulaCell(cell);
                             } catch (RuntimeException ex) {
-                                if (ex.getCause() instanceof NotImplementedFunctionException){
-                                    log.error("Runtime Exception is notImplementedFunction");
+                                Throwable cause = ex.getCause();
+                                String cellRef = new CellReference(cell).formatAsString();
+                                int rowIndex = cell.getRowIndex() + 1;
+                                String formula = getFormulaAsString(cell);
+                                switch (cause) {
+                                    case NotImplementedFunctionException nie -> {
+                                        errors.add(new DataImportError(rowIndex, List.of(String.format("Unsupported function '%s' at Sheet '%s'!%s. Formula: =%s", nie.getFunctionName(), sheetName, cellRef, formula))));
+                                    }
+                                    case FormulaParseException fpe -> {
+                                        errors.add(new DataImportError(rowIndex, List.of(String.format("Formula parse error at Sheet '%s'!%s. Formula: =%s. Reason: %s", sheetName, cellRef, formula, nullToEmpty(fpe.getMessage())))));
+                                    }
+                                    default -> {
+                                        errors.add(new DataImportError(rowIndex, List.of(String.format("Error evaluating at Sheet '%s'!%s. Formula: =%s. Reason: %s", sheetName, cellRef, formula, nullToEmpty(ex.getMessage())))));
+                                    }
                                 }
-                                log.error("RUntime exception cause is: " + ex.getCause().toString());
-                                errors.add(new DataImportError((cell.getRowIndex() + 1), List.of(String.format("Error evaluating at Sheet '%s'!%s. Formula: =%s. Reason: %s", sheetName, new CellReference(cell).formatAsString(), getFormulaAsString(cell), nullToEmpty(ex.getMessage())))));
+                                log.error("RUntime exception cause is: " + cause.toString());
```

Comment:
```suggestion
                                log.error("Excel import failed due to RuntimeException with cause: " + cause.toString());
```

typo. If we log it, we can directly make it more precise. If we can add a filename or similar (easily) would be even better.